### PR TITLE
Remove unused scheduler types

### PR DESCRIPTION
**Pull request by ahg-g**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

/kind cleanup

**Which issue(s) this PR fixes**:
Part of #85822

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/cc @alculquicondor

### CODE REVIEW

File: pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -26,9 +26,9 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/sets"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	api "k8s.io/kubernetes/pkg/apis/core"
```
Comment (alculquicondor) [inline pkg/scheduler/algorithm/predicates/predicates.go]: scheduler shouldn't depend in this package (See #84879). However, I can see other binaries using this.

Maybe we can use https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/apiserver/pkg/registry/generic/matcher.go#L25

Any suggestions @liggitt ?

File: pkg/scheduler/algorithm/predicates/predicates.go [hunk 2]
```diff
@@ -240,11 +240,9 @@ func PodFitsResourcesPredicate(pod *v1.Pod, podRequest *schedulernodeinfo.Resour
 // nodeMatchesNodeSelectorTerms checks if a node's labels satisfy a list of node selector terms,
 // terms are ORed, and an empty list of terms will match nothing.
 func nodeMatchesNodeSelectorTerms(node *v1.Node, nodeSelectorTerms []v1.NodeSelectorTerm) bool {
-	nodeFields := map[string]string{}
-	for k, f := range algorithm.NodeFieldSelectorKeys {
-		nodeFields[k] = f(node)
-	}
-	return v1helper.MatchNodeSelectorTerms(nodeSelectorTerms, labels.Set(node.Labels), fields.Set(nodeFields))
+	return v1helper.MatchNodeSelectorTerms(nodeSelectorTerms, labels.Set(node.Labels), fields.Set{
```
Comment (Huang-Wei) [inline pkg/scheduler/algorithm/predicates/predicates.go]: nit: 
```suggestion
	return v1helper.MatchNodeSelectorTerms(nodeSelectorTerms, node.Labels, fields.Set{
```

File: pkg/scheduler/algorithm/predicates/predicates.go [hunk 3]
```diff
@@ -240,11 +240,9 @@ func PodFitsResourcesPredicate(pod *v1.Pod, podRequest *schedulernodeinfo.Resour
...
+		api.ObjectNameField: node.Name,
```
Comment (alculquicondor) [inline pkg/scheduler/algorithm/predicates/predicates.go]: :thinking: It looks like we haven't actually documented that we only support "metadata.name" when using field selectors.

Comment (liggitt) [reply pkg/scheduler/algorithm/predicates/predicates.go]: API validation limits it. We can update the API documentation.

File: pkg/scheduler/algorithm/predicates/predicates.go (hunk 1 above)
Comment (liggitt) [reply pkg/scheduler/algorithm/predicates/predicates.go]: I would construct the fieldset in place as `"metadata.name": node.Name`

File: pkg/scheduler/algorithm/predicates/predicates.go (hunk 2 above)
Comment (ahg-g) [reply pkg/scheduler/algorithm/predicates/predicates.go]: done.

File: pkg/scheduler/algorithm/predicates/predicates.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/algorithm/predicates/predicates.go]: done.

**Pull request merged**